etna/datasets/utils.py

```diff
@@ -243,30 +239,30 @@ def get_level_dataframe(
     if num_target_level_segments != mapping_matrix.shape[0]:
         raise ValueError("Number of target level segments do not match mapping matrix number of columns!")
 
-    df = df.loc[:, pd.IndexSlice[source_level_segments, target_names]]
-
-    source_level_data = df.values  # shape: (t, num_source_level_segments * num_target_names)
+    source_level_data = df.loc[
+        pd.IndexSlice[:], pd.IndexSlice[source_level_segments, column_names]
```

Comment:
Here we go again with this cursed indexing...

Reply:
May be we should mark it somehow to find it later.

Reply:
We can, but here it is important that segments should be in the specific order